test/src/userobjects/PeriodicNodeMapTester.C

```diff
@@ -0,0 +1,92 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PeriodicNodeMapTester.h"
+#include "NonlinearSystemBase.h"
+
+registerMooseObject("MooseTestApp", PeriodicNodeMapTester);
+
+template <>
+InputParameters
+validParams<PeriodicNodeMapTester>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addClassDescription("Test MooseMesh::buildPeriodicNodeMap()");
+  params.addCoupledVar("v", "coupled variable (should be periodic)");
+  return params;
+}
+
+PeriodicNodeMapTester::PeriodicNodeMapTester(const InputParameters & parameters)
+  : ElementUserObject(parameters),
+    _v_var(coupled("v")),
+    _dim(_mesh.dimension()),
+    _perf_buildmap(registerTimedSection("buildPeriodicNodeMap", 1))
+{
+  // collect mesh periodicity data
+  for (unsigned int i = 0; i < _dim; ++i)
+  {
+    if (!_mesh.isRegularOrthogonal() || !_mesh.isTranslatedPeriodic(_v_var, i))
+      paramError("v", "Variable must be periodic in all directions");
```

Comment:
The recover test failed here. Either `_mesh.isRegularOrthogonal()` or `_mesh.isTranslatedPeriodic(_v_var, i)` does not return the correct value after recover.

Reply:
#11898